source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -236,8 +244,21 @@ void Filter::populateRateLimitDescriptors(const Router::RateLimitPolicy& rate_li
   }
 }
 
-void Filter::populateResponseHeaders(Http::HeaderMap& response_headers) {
+void Filter::populateResponseHeaders(Http::HeaderMap& response_headers,
+                                     bool overwrite_content_type) {
```

Comment:
I think it would be clearer if we renamed this parameter to reflect that we're sending a local response vs. not.  In the local case we treat the content-type in the ratelimiter response, as authoritative.

Let's document this behavior in the `RateLimitResponse` protobuf (I think it can go in the comment for `body`).

Reply:
Will do